lesson_15/hw_15.py

```diff
@@ -12,4 +12,110 @@
+        if self.angle_ab + self.angle_bc != 180:
+            raise ValueError("Сума суміжних кутів ромба повинна бути 180 градусів")
+        if self.angle_ab == 90 and self.angle_bc == 90:
+            raise ValueError("Кути ромба не можуть бути 90 градусів, бо тоді він переходе у прямокутник")
+
+    def __str__(self):
+        return f"Ромб має сторони довжиною {self.side_a}см та кути А та С {self.angle_ab} градусів, " \
+               f"а кути В та D мають {self.angle_bc} градусів"
+
+    '''Подумав, що так як написані мною умови не дадуть змінити один з кутів, бо почнеться перевірка та дасть помилку.
+    Тож додав сеттер, який може змінити обидва кути.'''
+    def set_angles(self, angle_ab, angle_bc):
+        super().__setattr__("angle_ab", angle_ab)
+        super().__setattr__("angle_bc", angle_bc)
+        self.check_angles()
+
+
+
+'''2 варіант классу. Спираючись на геометричні властивості ромбу, нам не потрібен ддрцгий кут у атрибутах,
+бо ми одразу можемо його обчислити выднявши заданий від 180. Що потім сильно спростить код та перевірки в ньому.
+Також спростить функціонал зміни його властивостей'''
+
+class Rhombus_2:
+    def __init__(self, side_a, angle_ab):
+        self.side_a = side_a
+        self.set_angles(angle_ab)
+
+    def __setattr__(self, name, value):
+        if name == "side_a":
+            if not isinstance(value, (int, float)):
+                raise TypeError("Довжина сторони повинна бути числом більше 0")
+            if value <= 0 or not value:
+                raise ValueError("Довжина сторони повинна бути числом більше 0")
+        elif name == "angle_ab":
+            if not isinstance(value, (int, float)):
+                raise TypeError("Значення angle_ab повинно бути більше 0 та менше 180")
+            if not (0 < value < 180) or not value:
+                raise ValueError("Значення angle_ab повинно бути більше 0 та менше 180")
+            if value == 90:
+                raise ValueError("Кути ромба не можуть бути 90 градусів, бо тоді він переходе у прямокутник")
+            super().__setattr__("angle_bc", 180 - value)
+            super().__setattr__("angle_cd", value)
+            super().__setattr__("angle_da", 180 - value)
+        super().__setattr__(name, value)
+
+    def set_angles(self, angle_ab):
+        self.angle_ab = angle_ab  # Здесь автоматически обновятся все углы через __setattr__
+
+    def __str__(self):
+        return f"Ромб має сторони довжиною {self.side_a}см та кути А та С {self.angle_ab} градусів, " \
+               f"а кути В та D мають {self.angle_bc} градусів"
+
+
+
+# А як на мене, найкращій та компактніший варіант всеж таки з ініт.
+
+class Rhombus_3:
+    def __init__(self, side_a, angle_ab):
+        if not isinstance(side_a, (int, float)) or side_a <= 0:
+            raise ValueError("Довжина сторони повинна бути числом більше 0")
+        self.side_a = side_a
+        self.set_angles(angle_ab)
+
+    def set_angles(self, angle_ab):
```

Comment:
```suggestion
    def set_angles(self, angle_ab):
```
це ж явно пропертя сюди сетер аж проситься